File: frontend/src/services/emrService.js
/**
 * Tuấn - HE204215
 * 
 * Chứa các API call liên quan đến quản lý Bệnh án điện tử (EMR).
*/
import axiosClient from '../api/axiosClient'

export const emrService = {
  // Hàm lưu bệnh án (tạo mới hoặc cập nhật bệnh án)
  saveEMR: (data) =>
    axiosClient.post('/v1/emr', data),

  // Hàm lấy chi tiết bệnh án dựa theo ID của lịch hẹn
  getByAppointment: (appointmentId) =>
    axiosClient.get(`/v1/emr/appointment/${appointmentId}`),

  // Hàm lấy lịch sử khám bệnh của một bệnh nhân cụ thể
  getPatientHistory: (patientId) =>
    axiosClient.get(`/v1/emr/patient/${patientId}/history`),

  // Hàm lấy danh sách tất cả hồ sơ bệnh án đã hoàn thành
  getCompletedList: () =>
    axiosClient.get(`/v1/emr/completed`),

  // Hàm lấy lịch sử bệnh án của chính bệnh nhân đang đăng nhập
  getLoggingInPatientHistory: () =>
    axiosClient.get(`/v1/emr/history`),

  getById: (id) =>
    axiosClient.get(`/v1/emr/${id}`),

  getAllList: () =>
    axiosClient.get(`/v1/emr/all`),
}

File: backend/src/main/java/com/ecms/service/EMRService.java
/** 
 * Author: Tuấn - HE204215
 * 
 * Interface định nghĩa các nghiệp vụ liên quan đến quản lý Hồ sơ bệnh án điện tử (EMR).
*/
package com.ecms.service;

import com.ecms.dto.request.EMRRequest;
import com.ecms.dto.response.EMRResponse;

import java.util.List;

public interface EMRService {

    // Lưu thông tin hồ sơ bệnh án (tạo mới hoặc cập nhật bệnh án hiện có)
    EMRResponse saveEMR(EMRRequest request);

    EMRResponse getById(Long id);

    // Lấy chi tiết hồ sơ bệnh án thông qua ID của lịch hẹn
    EMRResponse getByAppointmentId(Long appointmentId);

    // Lấy danh sách lịch sử khám bệnh của một bệnh nhân cụ thể
    List<EMRResponse> getPatientHistory(Long patientId);

    // Lấy danh sách tất cả lịch sử bệnh án đã hoàn thành
    List<EMRResponse> getCompletedList(Long doctorId);

    // Lấy toàn bộ hồ sơ bệnh án trong hệ thống (mọi trạng thái)
    List<EMRResponse> getAllList();
}
